src/components/node/DeleteNode.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import Button from '../common/Button';
+import { CommandResponse } from '../../hooks/useNodeManagement';
+import { DeleteNodeContainer, DeleteNodeText, ErrorMessage } from '../../styles/DeleteNodeStyles';
+
+interface DeleteNodeProps {
+  handleDeleteNode: () => Promise<CommandResponse>;
+  onCancel: () => void;
+  onDeleteSuccess: () => void;
+}
+
+const DeleteNode: React.FC<DeleteNodeProps> = ({ handleDeleteNode, onCancel, onDeleteSuccess }) => {
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  const handleDelete = async () => {
+    try {
```

Comment:
```suggestion
    setIsLoading(true); // disable delete button
    try {
    .....
    setIsLoading(false); // process finished
```

Can't test yet but if this component is not popup and stays open it should have this disable options